Float.Core/UX/CoordinatorParent.cs

```diff
@@ -117,6 +123,33 @@ public override string ToString()
             return $"[{GetType()}, Children: {string.Join(",", childCoordinators)}]";
         }
 
+        /// <summary>
+        /// Handles the finish request.
+        /// </summary>
+        /// <param name="coordinator">The calling coordinator.</param>
+        /// <param name="eventArgs">The event args.</param>
+        /// <returns>A value indicating whether this finished.</returns>
+        public override bool HandleFinishRequested(ICoordinator coordinator, EventArgs eventArgs)
+        {
+            var didFinish = true;
+
+            waitingToFinishEventArgs = eventArgs;
+
+            foreach (var eachChild in ChildCoordinators)
+            {
+                if (eachChild is Coordinator childCoordinator)
+                {
+                    var finished = childCoordinator.HandleFinishRequested(childCoordinator, eventArgs);
```

Comment:
It is not the child coordinator that was requested to finish, it was the parent.
```suggestion
                    var finished = childCoordinator.HandleFinishRequested(this, eventArgs);
```